assets/js/controller.js

```diff
@@ -0,0 +1,93 @@
+import {
+    view,
+    model
+} from './app.js';
+
+export default class Controller {
+    constructor() {
+        this.gameStatus = 0;
+        this.tilesOnBoard;
+        this.gameType;
+        this.initY;
+        this.initX;
+        this.startGameTime;
+        this.EndGameTime;
+        this.minSwipeDistance = 20;
+    }
+
+    startGame() {
+        this.tilesOnBoard = 0;
+        this.gameStatus = 1;
+        model.clearBoard();
+        view.clearBoard();
+        !localStorage.getItem(`bestScore${this.gameType}`) ? localStorage.setItem(`bestScore${this.gameType}`, 0) : null;
+        model.initGame(this.gameType);
+        view.initGame(this.gameType);
+        this.startGameTime = performance.now();
+        view.addTile(model.addTile());
+        view.addTile(model.addTile());
+    }
+
+    onKeyPress(key) {
+        (key === 'Escape' && view.popupVisible) ? view.togglePopup() : null;
+        if (!key.includes('Arrow') || (!this.gameStatus)) return;
+        view.moveTiles(model.moveTiles(key));
+        model.canAddTile ? view.addTile(model.addTile()) : null;
+        if (this.tilesOnBoard === (this.gameType * this.gameType)) {
+            let isContinued = model.checkIfPossibleMerge();
+            !isContinued ? this.endGame() : null;
+        }
+    }
+
+    onTouch(type, eX, eY) {
+        if (type === 'touchend') {
+            let key;
+            if (Math.abs(Math.round(eY - this.initY)) < this.minSwipeDistance && Math.abs(Math.round(eX - this.initX)) < this.minSwipeDistance) return;
+            let angle = (Math.atan2(Math.round(eY - this.initY), (Math.round(eX - this.initX))));
+            angle *= 180 / Math.PI;
+            angle = Math.round(angle);
+            if (angle === 0) return;
+            else {
+                if (angle >= -135 && angle < -45) {
+                    key = 'ArrowUp';
+                } else if (angle >= -45 && angle < 45) {
+                    key = 'ArrowRight';
+                } else if (angle >= 45 && angle < 135) {
+                    key = 'ArrowDown';
+                } else {
+                    key = 'ArrowLeft';
+                }
+            }
+            view.moveTiles(model.moveTiles(key));
+            model.canAddTile ? view.addTile(model.addTile()) : null;
+            if (this.tilesOnBoard === (this.gameType * this.gameType)) {
+                let isContinued = model.checkIfPossibleMerge();
+                !isContinued ? this.endGame() : null;
+            }
+        }
+        this.initY = eY;
+        this.initX = eX;
+    }
+
+    stopGame() {
+        this.gameStatus = 0;
```

Comment:
Najlepiej stworzyć plik `game-statuses.js`, który będzie wyglądałtak
```
export const gameStatuses {
  stopped: 0,
  costam: 1,
  costam2: 2
}
```
i posługiwać się w kodzie konkretnymi nazwami, np. `this.gameStatus = gameStatuses.stopped`